contrib/pam_zfs_key/pam_zfs_key.c

```diff
@@ -749,6 +778,7 @@ pam_sm_open_session(pam_handle_t *pamh, int flags,
 		    "Cannot zfs_mount when not being root.");
 		return (PAM_SUCCESS);
 	}
+	(void) session_roundtrip_check(pamh);
```

Comment:
What's the point of returning a tristate if the only user just void-casts it? The documentation says warn, and it just warns, so why not just warn?
```c
static void
session_roundtrip_check(pam_handle_t *pamh)
{
	const char *service;
	if (pam_get_item(pamh, PAM_SERVICE, (const void **)&service)
	    != PAM_SUCCESS)
		pam_syslog(pamh, LOG_NOTICE, "Unable to identify PAM service");
	else if (strcmp("systemd-user", service) == 0)
		pam_syslog(pamh, LOG_WARNING,
		    "Key may not be unloaded because of "
		    "https://github.com/systemd/systemd/issues/8598");
}
```

Reply:
I hoped to build on this change and add functionality based around the return value.  However, I think those changes could generate a more vibrant discussion.  Since this function provides value even in isolation, I decided to submit it separately.

Regardless, it doesn't seem to matter; both forms compile to the same binary on my system.